src/workerd/io/tracer.c++

```diff
@@ -196,14 +196,9 @@ void WorkerTracer::addLog(const tracing::InvocationSpanContext& context,
 }
 
 void WorkerTracer::addSpan(tracing::CompleteSpan&& span) {
-  // This is where we'll actually encode the span.
-  if (pipelineLogLevel == PipelineLogLevel::NONE) {
-    return;
-  }
-
-  // Note: spans are not available in the buffered tail worker, so we don't need an exceededSpanLimit
-  // variable for it and it can't cause truncation.
-  auto& tailStreamWriter = KJ_UNWRAP_OR_RETURN(maybeTailStreamWriter);
+  // This is where we'll actually encode the span. This function should never be invoked if STW is
+  // inactive as span tracing is only used in STW.
+  auto& tailStreamWriter = KJ_ASSERT_NONNULL(maybeTailStreamWriter);
```

Comment:
this is intended as additional hardening, using KJ_UNWRAP_OR_RETURN could cause us to miss out on span tracing being set up when it is not being used – I'm highly confident that we're already avoiding this though.